SDL_GameControllerHasRumble is undefine

In src/game_controller.cpp line 324
SDL_GameControllerHasRumble is undefine
Fail to build joy

Same problem for me when building ROS Humble from source in Ubuntu 20.04 Focal (a [Tier 3 platform](https://docs.ros.org/en/humble/Installation/Alternatives/Ubuntu-Development-Setup.html#system-requirements)):

```
[ 50%] Building CXX object CMakeFiles/game_controller.dir/src/game_controller.cpp.o
/opt/ros/humble/src/joystick_drivers/joy/src/game_controller.cpp: In member function ‘void joy::GameController::handleControllerDeviceAdded(const SDL_ControllerDeviceEvent&)’:
/opt/ros/humble/src/joystick_drivers/joy/src/game_controller.cpp:319:7: error: ‘SDL_GameControllerHasRumble’ was not declared in this scope; did you mean ‘SDL_GameControllerRumble’?
  319 |   if (SDL_GameControllerHasRumble(game_controller_)) {
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      |       SDL_GameControllerRumble
make[2]: *** [CMakeFiles/game_controller.dir/build.make:63: CMakeFiles/game_controller.dir/src/game_controller.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:167: CMakeFiles/game_controller.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

Installed versions:
```
$ dpkg -l | grep libsdl2
ii  libsdl2-2.0-0:amd64                                 2.0.10+dfsg1-3                        amd64        Simple DirectMedia Layer
ii  libsdl2-dev:amd64                                   2.0.10+dfsg1-3                        amd64        Simple DirectMedia Layer development files
```
and `sdl2_vendor` selects the system version by default.

If this is the only detail, then I agree with @mor2code that the `joy` package should probably be kept backwards-compatible by either using the old API or by compiling conditionally (the version can be checked by including `SDL2/SDL_version.h`).

I don't see how https://github.com/ros-drivers/joystick_drivers/pull/270 would solve the problem (and it does not) as long as `SDL2_FOUND` is true and we use the system version. I tested that patch, without success.

The pull request that introduced the new API: https://github.com/ros-drivers/joystick_drivers/pull/258

This repository also has a [`humble-devel`](https://github.com/ros-drivers/joystick_drivers/tree/humble-devel) branch that is not affected, but the [rosdistro index](https://github.com/ros/rosdistro/blob/f6853594cef801828e29cc90c21c3ac6cdb954af/humble/distribution.yaml#L2632) and https://index.ros.org/p/joy/#humble point to the `ros2` branch for Humble?

--- a/joy/src/game_controller.cpp
+++ b/joy/src/game_controller.cpp
@@ -315,10 +315,14 @@ void GameController::handleControllerDeviceAdded(const SDL_ControllerDeviceEvent
     joy_msg_.axes.at(i) = convertRawAxisValueToROS(state);
   }
 
+#if SDL_VERSION_ATLEAST(2, 0, 18)
   const char * has_rumble_string = "No";
   if (SDL_GameControllerHasRumble(game_controller_)) {
     has_rumble_string = "Yes";
   }
+#else
+  const char * has_rumble_string = "Unknown";
+#endif
 
   RCLCPP_INFO(
     get_logger(), "Opened game controller: %s,  deadzone: %f, rumble: %s",

